### PR TITLE
Hue Python3.8 upgrade

### PR DESCRIPTION
**Pull request by ranade1**
## What changes were proposed in this pull request?

- build
- oozie
- solr
- notebook
- filemanager
- backend

## How was this patch tested?

- tested on CentOS7/Postgres10 and other systems.

Please review [Hue Contributing Guide](https://github.com/cloudera/hue/blob/master/CONTRIBUTING.md) before opening a pull request.

### CODE REVIEW

**Review by wing2fly — APPROVED** [review]

File: desktop/libs/hadoop/src/hadoop/fs/hadoopfs.py [hunk 1]
```diff
@@ -227,28 +232,67 @@ def _copy_dir(self, local_dir, remote_dir, mode=0o755):
...
 
-  def _copy_file(self, local_src, remote_dst, chunk_size=1024 * 1024 * 64):
-    if os.path.isfile(local_src):
-      if self.exists(remote_dst):
-        LOG.info(_('%(remote_dst)s already exists. Skipping.') % {'remote_dst': remote_dst})
-        return
+  def _copy_binary_file(self, local_src, remote_dst, chunk_size):
+    src = open(local_src, mode="rb")
+    try:
+      try:
```
Comment (Harshg999) [inline desktop/libs/hadoop/src/hadoop/fs/hadoopfs.py:238]: Why 2 try block? Can we add `finally` in one try block?

**Review by Harshg999 — CHANGES_REQUESTED** [review]

File: desktop/core/requirements.txt [hunk 2]
```diff
@@ -40,11 +39,13 @@ lockfile==0.12.2
 lxml==4.6.5
 Mako==1.1.4
 Markdown==3.1
+mysqlclient==2.1.0
```
[the remark this answers is not in the document]
Comment (JohanAhlen) [reply desktop/core/requirements.txt:42]: Yes, this specific version indeed. For 1.2.x or earlier it's optionally CNRI which works for us.

File: desktop/core/requirements.txt [hunk 3]
```diff
@@ -1,9 +1,8 @@
 apache-ranger==0.0.3
+requests-gssapi==1.2.3
 asn1crypto==0.24.0
 avro-python3==1.8.2
-Babel==2.9.1
-boto==2.46.1
-# boto3==1.17.41  # Last Py2 is 1.17
+Babel==2.5.3
```
[the remark this answers is not in the document]
Comment (ranade1) [reply desktop/core/requirements.txt:5]: @Harshg999 I think there was some problem with Babel on 2.9.1 version with Python3.8. I don't remember now. Once this change(with Babel==2.5.3) is in we will upgrade Babel to required version.

File: desktop/core/requirements.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (ranade1) [reply desktop/core/requirements.txt:42]: removing mysqlclient as LGPLv2 is not permissible enough.

**Review by wing2fly — APPROVED** [review]

**Pull request merged**